### PR TITLE
Add network info to show_node and add documentation for REST API

### PR DESCRIPTION
**Pull request by zenhack**
This modifies the show_node api call as suggested by @henn's comments in #453.

### CODE REVIEW

**Comment by zenhack** [comment]
This includes #455 and #468, so those should be merged first.

File: haas/api.py [hunk 1]
```diff
@@ -822,6 +813,9 @@ def show_node(nodename):
         'free': node.project_id is None,
         'nics': [{'label': n.label,
                   'macaddr': n.mac_addr,
+                  'networks': dict([(attachment.channel,
```
Comment (henn) [inline haas/api.py]: For human parsing, it might be helpful to return everything sorted.

Comment (zenhack) [inline haas/api.py]: It's a dictionary --- it's inherently unordered.

Comment (henn) [inline haas/api.py]: I was thinking of something like https://stackoverflow.com/questions/9001509/how-can-i-sort-a-dictionary-by-key

Comment (zenhack) [inline haas/api.py]: So I played around with this and it impircally seems to work, but I have a strong objection to this: There's nothing in the json module's interface that says it respects the ordering, and it isn't required to --- json objects are unordered. We'd be relying on undocumented implementation details, and we should *never* do that.

Comment (henn) [inline haas/api.py]: I understand it's not required to. As a human interactive user, it's just easier for me to work with a list that is sorted; especially in operations where I think something is connected a certain way and I just want to verify that.

Comment (henn) [inline haas/api.py]: Looks like [json.dump()](https://docs.python.org/2.7/library/json.html#json.dump) supports this with `sort_keys=True`!

**Comment by zenhack** [comment]
@henn, I've addressed your comment.

**Comment by henn** [comment]
:+1: (which is `:+1:`)

**Comment by zenhack** [comment]
@henn, we decided to leave the waiting on reviewer label on a pr even after adding the waiting on +2 (the reasoning being that github doesn't give you a logical or, so removing it makes searching for things that need review more annoying). Just fyi.

**Comment by gsilvis** [comment]
We definitely want to merge something like this, I think... but it has merge conflicts, and the diff looks MUCH larger than it needs to be.  I think one of the other PRs this depended on got changed somehow, so this'll take some untangling.

**Comment by henn** [comment]
This works was picked up in #537. Thank you for your contribution!

**Pull request closed without merging**